tasty-query/shared/src/main/scala/tastyquery/Symbols.scala

```diff
@@ -655,12 +618,37 @@ object Symbols {
       */
     final def getPackageDecl(name: SimpleName)(using Context): Option[PackageSymbol] =
       /* All subpackages are created eagerly when initializing contexts,
-       * so we can use getDeclInternal here.
+       * so we can directly access myDeclarations here.
        */
-      getDeclInternal(name).collect { case pkg: PackageSymbol =>
+      myDeclarations.get(name).collect { case pkg: PackageSymbol =>
         pkg
       }
     end getPackageDecl
+
+    private[Symbols] final def addDecl(decl: Symbol): Unit =
+      assert(!myDeclarations.contains(decl.name), s"trying to add a second entry $decl for name ${decl.name} in $this")
+      myDeclarations(decl.name) = decl
+
+    final def getDecls(name: Name)(using Context): List[Symbol] =
+      getDecl(name).toList
+
+    private final def ensureRootsInitialized()(using Context): Unit =
+      if !rootsInitialized then
+        ctx.classloader.scanPackage(this)
+        rootsInitialized = true
+
+    final def getDecl(name: Name)(using Context): Option[Symbol] =
+      myDeclarations.get(name).orElse {
+        ensureRootsInitialized()
```

Comment:
I would suggest making `ensureRootsInitialized` return boolean, then only include the fallback when the package was just scanned now

Reply:
That wouldn't be correct. If you first do `getDecl("foo")`, roots will be initialized. But then you do `getDecl("bar")`, and then `ensureRootsInitialized()` will do nothing, but you *still* need to try `loadRoot(this, "bar")`.

Reply:
yes right